modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1658,6 +1658,32 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 		}
 	}
 
+	if (p_function->is_vararg()) {
+		resolve_parameter(p_function->rest_parameter);
+		if (p_function->rest_parameter->datatype_specifier != nullptr) {
+			GDScriptParser::DataType specified_type = p_function->rest_parameter->get_datatype();
+			if (specified_type.kind != GDScriptParser::DataType::BUILTIN || specified_type.builtin_type != Variant::ARRAY ||
+					(specified_type.has_container_element_type(0) && !specified_type.get_container_element_type(0).is_variant())) {
+				push_error(vformat(R"(The rest parameter type must be "Array", but "%s" is specified.)", specified_type.to_string()), p_function->rest_parameter->datatype_specifier);
```

Comment:
Do we really want "rest parameter" as user facing name? I feel like dropping this name without context does not really make it clear what is talked about. I think vararg parameter would be easier for users to understand. (For reference were is the name taken from?)

Reply:
> For reference were is the name taken from?

[Rest parameters - JavaScript | MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters)

Reply:
Every time you think you know javascript :sweat_smile:

I still think vararg would be a better name, since it is already used in the docs for print.